verl/trainer/ppo/ray_trainer.py

```diff
@@ -150,13 +151,18 @@ def _check_resource_available(self):
                 )
 
 
-def apply_kl_penalty(data: DataProto, kl_ctrl: core_algos.AdaptiveKLController, kl_penalty="kl"):
+def apply_kl_penalty(data: DataProto, kl_ctrl: core_algos.AdaptiveKLController, kl_penalty="kl", multi_turn=False):
     responses = data.batch["responses"]
     response_length = responses.size(1)
     token_level_scores = data.batch["token_level_scores"]
     batch_size = data.batch.batch_size[0]
-    attention_mask = data.batch["attention_mask"]
-    response_mask = attention_mask[:, -response_length:]
+
+    if multi_turn:
```

Comment:
Shall we directly introduce loss_mask inside the data. By default, it should be equal to attention_mask. Then, replace the attention_mask here with loss_mask. This will simplify the code a lot without introducing too many branches.